[dont merge]  branch page some enhancements (-> PRs #7602 #7603 #7604)

6543 opened the pull request:
is part of the issue "branch view some enhancements #7454"

* [x]  all buttons on the right side should be with border
* [x]  find another way to layout the button correctly that does at least not rely on absolute positioning
* [x]  move button info into buttons
* [x]  delete table header
* [x] move inline css to less

6543 commented:
![screenshot](https://i.ibb.co/Wgr6MKq/Bildschirmfoto-zu-2019-07-18-19-22-31.png)

6543 commented:
Error: 

![](https://ipfs.io/ipfs/QmZC5sy8FZ2cBzWqWZw2mquCE8JpretLGsugoAAcCPGfnD) ![](https://ipfs.io/ipfs/QmcnsgJHy2eeARXN8UGv72ztMbrqMDNgJMngnRoB9d5JGY)

lunny commented:
duplicated protect button.

gary-kim commented:
> duplicated protect button.

That's not from this PR. I added the icon on the left so users without write access can also see if the branch is protected. 
Maybe instead of the protected icon, we should just get rid of the delete branch button if the branch is protected?

6543 commented:
@gary-kim 1 vote for your solution, and what do you think of the download buton in gereral?
if they are nice i'll try to fix the menue issue else ill del this pull req ...

gary-kim commented:
It seems to have broken for me. Using Firefox 67.
![Screenshot from 2019-07-19 13-58-00](https://user-images.githubusercontent.com/47195730/61513201-e66fb500-a9eb-11e9-8f27-8d588ca2a624.png)

6543 commented:
It is wip, i need sugestions how to do, sory i'm a css newby

6543 commented:
@gary-kim  test it now,

the ownly thing thats left is the download menue is not in the front - do you know how to get it to on top of the table?

6543 commented:
Preview:
![](https://ipfs.io/ipfs/QmcUrLK9JbzwrAsPwFZkg65ECbGFjD6z2LLxYiNncr5dvw)

6543 commented:
> > duplicated protect button.
> 
> That's not from this PR. I added the icon on the left so users without write access can also see if the branch is protected.
> Maybe instead of the protected icon, we should just get rid of the delete branch button if the branch is protected?

i moved the protect symbol to the delete line, all symbols are in one row. if this is ok, we should rename the colum header or delete the table header because it isn't very usefull at all

6543 commented:
# Preview
## before login:
![before](https://ipfs.io/ipfs/QmXuma8SHAULwF6KaPb4YG3CmvvFtLCpJ37QwKXx5i4e6y)
## after login:
![after](https://ipfs.io/ipfs/QmcrUkfASX7SHyrEvrYaxD69EgHz6FoJfLhgvcLQwxMoEw)

techknowlogick commented on templates/repo/branch/list.tmpl:
```diff
@@ -92,8 +99,19 @@
 									</td>
+									<td class="right aligned">
+									  {{if not .IsDeleted}}
+											<div class="ui basic jump dropdown icon button poping up" style="position: absolute; margin-left: -38px; margin-top: -18px;" data-content="{{$.i18n.Tr "repo.download_archive"}}" data-variation="tiny inverted" data-position="top right">
```
Is it possible for you to move this (and all other) inline CSS, that you've added, into the less files?

6543 replied:
OK

6543 replied:
i tryed it - dont get it jet :(
there is only one inline: `style="position: absolute; margin-left: -38px; margin-top: -18px;"`
wich only afects  `class="ui basic jump dropdown icon button poping up"`

6543 replied:
i looked round about 1h, do you have any sugestions where i can read about less

richmahn replied:
The less files are in /public/less and then are generated by `make generate-stylesheets` to put all of them into /public/css/index.css (generated). I assume you'd put your CSS additions in /public/less/_repository.less

richmahn replied:
LESS (not sure if this is the reason for its name, but how I remembe rit) lets you type `less` than type typical CSS files since you can put selectors within selectors, a hierarchy of sort, and is a little more dynamic and has some code sytnax. Not sure if you were asking how to know more about what less is, or just read about how we do it in Gitea (my last comment will help there).

6543 replied:
@richmahn thanks!!

6543 replied:
less itself wasnt the problem, it is very intuetiv.   the `/public/less/_repository.less` was the problem.
because of an related pull i used `/public/less/_explore.less` the whole time.

for now it shuld work but in the future i rely have to now how **gitea**s less files are structured

6543 replied:
@techknowlogick i added a new class caled dlmenu is this ok?

6543 commented:
@techknowlogick review?

6543 commented:
@techknowlogick review/approve?

silverwind commented on public/less/_repository.less:
```diff
@@ -1233,6 +1233,11 @@
+            position: absolute;
+            margin-left: -38px;
+            margin-top: -18px;
+        }
```
This is rather ugly and such absolute positioned pixel hacks are bound to break eventually. Please find another way to layout the button correctly that does at least not rely on absolute positioning.

silverwind commented:
Besides my comment above, I think all buttons on the right side should either be borderless or with border. I'm leaning towards bordered.

6543 commented:
Ok

lafriks commented on options/locale/locale_de-DE.ini:
```diff
@@ -1400,6 +1400,8 @@ branch.deleted_by=Von %s gelöscht
 branch.restore_success=Branch „%s“ wurde wiederhergestellt.
 branch.restore_failed=Wiederherstellung des Branches „%s“ fehlgeschlagen.
 branch.protected_deletion_failed=Branch „%s“ ist geschützt und kann nicht gelöscht werden.
+branch.download=Branch „%s“ herunterladen
```
Do not add translations directly in code, it needs to be done using crowdin later

6543 replied:
https://github.com/go-gitea/gitea/blob/master/CONTRIBUTING.md#translation got it

6543 commented:
@silverwind new Preview:

![](https://ipfs.io/ipfs/QmfJKka6jiLqQEY7VEnNkH5yMBt9QhVCNx4tW9SWpZZHe1)

![](https://ipfs.io/ipfs/QmbayRUfN9bTxx7ttfF4rhd7KtHiG37D3j1cHCyX3Xx5WU)

6543 commented:
Current State: https://github.com/go-gitea/gitea/pull/7520#issuecomment-513991651

gary-kim commented on options/locale/locale_en-US.ini:
```diff
@@ -1407,6 +1407,8 @@ branch.deleted_by = Deleted by %s
 branch.restore_failed = Failed to restore branch '%s'.
 branch.protected_deletion_failed = Branch '%s' is protected. It cannot be deleted.
+branch.download=Download Branch „%s“
+branch.restore=Restore Branch „%s“
```
```suggestion
branch.restore = Restore Branch '%s'
```

gary-kim commented on options/locale/locale_en-US.ini:
```diff
@@ -1407,6 +1407,8 @@ branch.deleted_by = Deleted by %s
 branch.restore_success = Branch '%s' has been restored.
 branch.restore_failed = Failed to restore branch '%s'.
 branch.protected_deletion_failed = Branch '%s' is protected. It cannot be deleted.
+branch.download=Download Branch „%s“
```
```suggestion
branch.download = Download Branch '%s'
```
I think it'd be better to be consistent with how messages regarding branches refer to the branch names

6543 commented:
> I like how the download button looks now. There still seems to be a few things to be fixed up though.
> 
> Also, in your image, it seems like the right border of the `New Pull Request` button is cut off. Is that an issue unique to the Blink engine? It doesn't seem to be happening in Firefox.
> 
> EDIT: Oh, turns out the border being cut off happens when the screen is smaller then 1200px. Once you pass the breakpoint, the button border on the right side gets cut off. That should get fixed.

saw this bug too, but thought this sint relatet tho this PR would put it into another one.

6543 commented on templates/repo/branch/list.tmpl:
```diff
@@ -34,33 +43,20 @@
 										<s><a href="{{$.RepoLink}}/src/branch/{{.Name | EscapePound}}">{{.Name}}</a></s>
 										<p class="info">{{$.i18n.Tr "repo.branch.deleted_by" .DeletedBranch.DeletedBy.Name}} {{TimeSinceUnix .DeletedBranch.DeletedUnix $.i18n.Lang}}</p>
 									{{else}}
-										{{if .IsProtected}}
```
-> added to the todo list (top of PR)

6543 commented:
one suggestion: because this PR was originaly thought for the download button only:
is there to mouch redundant code? should i move the download button somehow into a seperate themplate?
beside the branch name the code is 3times the same:

list.tmpl A:
```
<td class="right aligned download-column">
  <div class="ui basic jump dropdown icon button poping up" data-content="{{$.i18n.Tr "repo.branch.download" ($.DefaultBranch|EscapePound)}}" data-variation="tiny inverted" data-position="top right">
    <i class="download icon"></i>
    <div class="menu">
      <a class="item" href="{{$.RepoLink}}/archive/{{EscapePound $.DefaultBranch}}.zip"><i class="octicon octicon-file-zip"></i> ZIP</a>
      <a class="item" href="{{$.RepoLink}}/archive/{{EscapePound $.DefaultBranch}}.tar.gz"><i class="octicon octicon-file-zip"></i> TAR.GZ</a>
    </div>
  </div>
</td>
```
list.tmpl B:
```
<div class="ui basic jump dropdown icon button poping up" data-content="{{$.i18n.Tr "repo.branch.download" (.Name | EscapePound)}}" data-variation="tiny inverted" data-position="top right">
  <i class="download icon"></i>
  <div class="menu">
    <a class="item" href="{{$.RepoLink}}/archive/{{.Name | EscapePound}}.zip"><i class="octicon octicon-file-zip"></i> ZIP</a>
    <a class="item" href="{{$.RepoLink}}/archive/{{.Name | EscapePound}}.tar.gz"><i class="octicon octicon-file-zip"></i> TAR.GZ</a>
  </div>
</div>
```
home.tmpl:
```
<div class="ui basic jump dropdown icon button poping up" data-content="{{.i18n.Tr "repo.download_archive"}}" data-variation="tiny inverted" data-position="top right">
  <i class="download icon"></i>
  <div class="menu">
    <a class="item" href="{{$.RepoLink}}/archive/{{EscapePound $.BranchName}}.zip"><i class="octicon octicon-file-zip"></i> ZIP</a>
    <a class="item" href="{{$.RepoLink}}/archive/{{EscapePound $.BranchName}}.tar.gz"><i class="octicon octicon-file-zip"></i> TAR.GZ</a>
  </div>
</div>
```

6543 commented:
@lafriks @gary-kim is this state now ready for a merge?
@gary-kim i think your sugesstions are new issues -> PR (https://github.com/go-gitea/gitea/issues/7577; https://github.com/go-gitea/gitea/issues/7576)

silverwind commented on templates/repo/branch/list.tmpl:
```diff
@@ -92,14 +91,25 @@
 											{{end}}
 										{{end}}
 									</td>
+									<td class="right alinged one wide download-column">
```
typo `alinged `?

silverwind commented:
I think the current implementation takes away too much space from the important left-aligned column because having the buttons inside individual fixed columns introduces unnecessary spacing. I'd put all three buttons in the same `tr` and make them look like this (remove `compact` class on the pull button):

![](https://user-images.githubusercontent.com/115237/61814988-8d6bab00-ae49-11e9-9b79-e227e8033eab.png)

Then right-align the diffstat and let the leftmost column take the remaining space.

silverwind commented:
Oh, and if you want to save even more space, change the pull button to a icon-only button.

6543 commented:
I'll split this pull up into seperate once !

6543 commented:
splitup is complete
